### ISSUE TITLE
Can't extend a8sidecar nginx.conf with stream socket context by adding a file to /etc/nginx/conf.d

### ISSUE THREAD

**Issue by greglanthier** [amalgam8/amalgam8#402]
I would like to use Amalgam8 for an application that is composed of web services as well as JMS, MongoDB, and JDBC servers.

IIUC it is not currently possible to proxy TCP or UDP sockets via Amalgam8 sidecars - presumably due to issues alluded to by [issues/4#issuecomment-245413281](https://github.com/amalgam8/amalgam8/issues/4#issuecomment-245413281).  I don't know if the Lua magic used by the a8sidecar can generate Nginx configuration for TCP and UDP sockets.

Theoretically we could extend the Nginx configuration used by the amalgam8/a8-sidecar:alpine Docker container to proxy TCP or UDP sockets on our own until the Lua thing is addressed, but when I tried I ran into the problems described in this [comment](https://github.com/amalgam8/amalgam8/issues/4#issuecomment-261077407).

> I have a ```mongo-proxy.conf``` file that looks like this:
>```
> stream {
>  upstream backend {
>    server <ip>:27017;
>  }
>  server {
>    listen 21212;
>    proxy_connect_timeout 1s;
>    proxy_timeout 1m;
>    proxy_pass backend;
>  }
>}
>```
>that I mount into `amalgam8/a8-sidecar:alpine` as `/etc/nginx/conf.d/mongo-proxy.conf` but at startup this fails since it looks like the default nginx.conf file loads `/etc/nginx/conf.d/*.conf` files into the http context.
>
>For example:
>
>mongo_sidecar    | time="2016-11-16T21:25:46Z" level=info msg="Starting NGINX" 
>mongo_sidecar    | time="2016-11-16T21:25:46Z" level=error msg="Failed to start NGINX service" error="exit status 1: nginx: [emerg] \"stream\" directive is not allowed here in /etc/nginx/conf.d/mongo_proxy.conf:1\n" 
>mongo_sidecar    | time="2016-11-16T21:25:46Z" level=warning msg="Registry listener f

One way to allow us to customise the Nginx configuration in the a8sidecar would be to do something like this in the [nginx.conf](https://github.com/amalgam8/amalgam8/blob/master/sidecar/nginx/conf/nginx.conf#L16-L33) configuration file:

```
http {
  ...
  include /etc/nginx/conf.d/*.http.conf;
  ...
}

stream {
  include /etc/nginx/conf.d/*.stream.conf;
}
```

I don't know very much about Nginx or its configuration best practices.  Would such a change lead to problems?

Alternatively would it make more sense for the `include` directive in the `http.conf` file to appear _outside_ the `http` context and rely on the user supplied files in `/etc/nginx/conf.d/*.conf` to include the contexts on their own?  I don't know if that option is even technically possible - if Nginx will combine the context declarations from multiple files into one overall declaration.

**Comment by rshriram** [comment]
@greglanthier can you try out the fix in PR #403 ? You would just have to overwrite the nginx.conf available as part of current sidecar distribution with the nginx.conf in #403 (this is temporary). And then simply add the mongo proxy config to your image or mount it in /etc/nginx/conf.d/mongo_proxy.stream.conf.

**Comment by greglanthier** [comment]
Thanks @rshriram!

I needed to append a `;` to the include directive [in your change](https://github.com/amalgam8/amalgam8/pull/403/commits/9e59ed8f37b821a20a55a3f779a3a061723ad400#diff-36893306ac07efcf3e19e651e57f3184R36).  For example:

```
 include /etc/nginx/conf.d/*.stream.conf
```

becomes

```
  include /etc/nginx/conf.d/*.stream.conf;
```

After that the Nginx instance in the `a8sidecar` image started up as expected when I mounted the following Nginx config file as a volume under the name `/etc/nginx/conf.d/mongo-proxy.stream.conf`:

```
upstream backend {
    server <ip>:27017;
}

server {
    listen 21212;
    proxy_connect_timeout 1s;
    proxy_timeout 1m;
    proxy_pass backend;
}
```

I'm going to assume this Nginx custom configuration step will only be a temporary measure until the tcp / udp endpoint types are configured automatically some how, is that correct?
